### PR TITLE
Filter in notifications drawer

### PR DESCRIPTION
**Pull request by CodyWMitchell**
[RHCLOUD-29014](https://issues.redhat.com/browse/RHCLOUD-29014)
- Enable filtering in the notifications drawer
- Fetch correct filters from `/api/notifications/v1/notifications/facets/bundles`

### CODE REVIEW

File: src/components/NotificationsDrawer/DrawerPanelContent.tsx [hunk 1]
```diff
@@ -38,6 +37,18 @@ import BulkSelect from '@redhat-cloud-services/frontend-components/BulkSelect';
 import axios from 'axios';
 import { Stack, StackItem } from '@patternfly/react-core/dist/dynamic/layouts/Stack';
 
+interface Bundle {
+  id: string;
+  name: string;
+  displayName: string;
+  children: null | Bundle[];
```
Comment (Hyperkid123) [inline src/components/NotificationsDrawer/DrawerPanelContent.tsx:44]: I think children can be optional `children?: Bundle[]`. Or is there is a specific need for the `null` value?

File: src/components/NotificationsDrawer/DrawerPanelContent.tsx [hunk 2]
```diff
@@ -124,7 +136,23 @@ const DrawerPanelBase = ({ innerRef }: DrawerPanelProps) => {
         );
       }
     };
+    const fetchFilterConfig = async () => {
```
Comment (Hyperkid123) [inline src/components/NotificationsDrawer/DrawerPanelContent.tsx:139]: Why is this function defined in `useEffect`?

Comment (CodyWMitchell) [reply src/components/NotificationsDrawer/DrawerPanelContent.tsx:139]: I was following how the permissions were fetched after `mounted` became true in the `fetchPermissions` method.

File: src/components/NotificationsDrawer/DrawerPanelContent.tsx (hunk 1 above)
Comment (CodyWMitchell) [reply src/components/NotificationsDrawer/DrawerPanelContent.tsx]: Good catch! I think you're right.

File: src/components/NotificationsDrawer/DrawerPanelContent.tsx (hunk 2 above)
Comment (Hyperkid123) [reply src/components/NotificationsDrawer/DrawerPanelContent.tsx:139]: I see. The pattern used before is wrong. The mounted variable should be in ref so we don't have to do this. Its something for a future refactoring I guess.

**Review by Hyperkid123 — APPROVED** [review]

**Pull request merged**